pandas/tests/plotting/frame/test_frame.py

```diff
@@ -2215,6 +2215,16 @@ def test_secondary_y(self, secondary_y):
                 assert ax.get_ylim() == (0, 100)
                 assert ax.get_yticks()[0] == 99
 
+    def test_sort_columns_deprecated(self):
+        df = DataFrame({"a": [1, 2], "b": [3, 4]})
+
+        with tm.assert_produces_warning(FutureWarning):
+            df.plot.box("a", sort_columns=True)
+            df.plot.box(sort_columns=False)
```

Comment:
I think you need an additional with block, this should pass when only one raises the warning

Reply:
Good catch!